MetaMask/metamask-mobile
# fix: allow signing on permissioned account

## **Description**

This PR enables the account the dapp is connected to be able to sign when using the deeplinking communication layer of the iOS native SDK. Currently the account can sign if it is both the connected to account and the selected account.

## **Manual testing steps**

1. Request connection to MM wallet from dapp
2. Choose any account other than the selected account as the account to connect to
3. Then sign a message. The wallet should be able to sign using the account the dapp is connected to, regardless of whether it's the selected account or not.

## **Screenshots/Recordings**



### **Before**

https://github.com/user-attachments/assets/4488ac44-4631-464b-bf74-9c55852c8e02

### **After**

https://github.com/user-attachments/assets/bea58455-c86e-4026-8f39-97e3aceaf3cd



## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
